src/Juvix/Compiler/Core/Language.hs

```diff
@@ -110,38 +142,51 @@ type Type = Node
 
 instance HasAtomicity Node where
   atomicity = \case
-    Var {} -> Atom
-    Ident {} -> Atom
-    Constant {} -> Atom
-    App {} -> Aggregate appFixity
-    BuiltinApp {..} | null _builtinArgs -> Atom
-    BuiltinApp {} -> Aggregate lambdaFixity
-    Constr {..} | null _constrArgs -> Atom
-    Constr {} -> Aggregate lambdaFixity
-    Lambda {} -> Aggregate lambdaFixity
-    Let {} -> Aggregate lambdaFixity
-    Case {} -> Aggregate lambdaFixity
-    Pi {} -> Aggregate lambdaFixity
-    Univ {} -> Atom
-    TypeConstr {} -> Aggregate appFixity
+    NVar {} -> Atom
```

Comment:
we probably want to have instances for each of the wrapper types and the instance for Node should just call those.
e.g.
```
instance HasAtomicity Constr where
  atomicity Constr {..} 
   | null _constrArgs -> Atom
   | otherwise -> Aggregate lambdaFixity

instance HasAtomicity Node where
  atomicity = \case 
     ....
     NCtr c -> atomicity c

Reply:
Done